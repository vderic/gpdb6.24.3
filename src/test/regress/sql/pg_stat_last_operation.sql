CREATE TABLE PG_STAT_LAST_OPERATION_TEST (foo int) DISTRIBUTED BY (foo);

SELECT classname, objname, schemaname, actionname, subtype 
FROM pg_stat_operations WHERE schemaname = 'public' 
AND objname = 'pg_stat_last_operation_test'
AND actionname = 'CREATE';

SELECT classname, objname, schemaname, actionname, subtype 
FROM pg_stat_operations WHERE schemaname = 'public' 
AND objname = 'pg_stat_last_operation_test'
AND actionname = 'TRUNCATE';

CREATE VIEW
	pg_stat_last_operation_testview_schema AS
SELECT lo.staactionname,
       lo.stasubtype,
       ns.nspname
FROM   pg_stat_last_operation lo
       join pg_class c
         ON lo.classid = c.oid
       join pg_namespace ns
         ON c.relname = 'pg_namespace'
            AND lo.objid = ns.oid;

insert into PG_STAT_LAST_OPERATION_TEST select generate_series(1,100);

truncate PG_STAT_LAST_OPERATION_TEST;

SELECT classname, objname, schemaname, actionname, subtype 
FROM pg_stat_operations WHERE schemaname = 'public' 
AND objname = 'pg_stat_last_operation_test'
AND actionname = 'TRUNCATE';

-- CREATE/DROP SCHEMA
CREATE SCHEMA mdt_schema;
SELECT * FROM pg_stat_last_operation_testview_schema WHERE nspname LIKE 'mdt_%';
DROP SCHEMA mdt_schema;
SELECT * FROM pg_stat_last_operation_testview_schema WHERE nspname LIKE 'mdt_%';
