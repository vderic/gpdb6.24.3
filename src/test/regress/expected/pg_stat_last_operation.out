CREATE TABLE PG_STAT_LAST_OPERATION_TEST (foo int) DISTRIBUTED BY (foo);
SELECT classname, objname, schemaname, actionname, subtype 
FROM pg_stat_operations WHERE schemaname = 'public' 
AND objname = 'pg_stat_last_operation_test'
AND actionname = 'CREATE';
 classname |           objname           | schemaname | actionname | subtype 
-----------+-----------------------------+------------+------------+---------
 pg_class  | pg_stat_last_operation_test | public     | CREATE     | TABLE
(1 row)

SELECT classname, objname, schemaname, actionname, subtype 
FROM pg_stat_operations WHERE schemaname = 'public' 
AND objname = 'pg_stat_last_operation_test'
AND actionname = 'TRUNCATE';
 classname | objname | schemaname | actionname | subtype 
-----------+---------+------------+------------+---------
(0 rows)

CREATE VIEW
	pg_stat_last_operation_testview_schema AS
SELECT lo.staactionname,
       lo.stasubtype,
       ns.nspname
FROM   pg_stat_last_operation lo
       join pg_class c
         ON lo.classid = c.oid
       join pg_namespace ns
         ON c.relname = 'pg_namespace'
            AND lo.objid = ns.oid;
insert into PG_STAT_LAST_OPERATION_TEST select generate_series(1,100);
truncate PG_STAT_LAST_OPERATION_TEST;
SELECT classname, objname, schemaname, actionname, subtype 
FROM pg_stat_operations WHERE schemaname = 'public' 
AND objname = 'pg_stat_last_operation_test'
AND actionname = 'TRUNCATE';
 classname |           objname           | schemaname | actionname | subtype 
-----------+-----------------------------+------------+------------+---------
 pg_class  | pg_stat_last_operation_test | public     | TRUNCATE   | 
(1 row)

-- CREATE/DROP SCHEMA
CREATE SCHEMA mdt_schema;
SELECT * FROM pg_stat_last_operation_testview_schema WHERE nspname LIKE 'mdt_%';
 staactionname | stasubtype |  nspname   
---------------+------------+------------
 CREATE        | SCHEMA     | mdt_schema
(1 row)

DROP SCHEMA mdt_schema;
SELECT * FROM pg_stat_last_operation_testview_schema WHERE nspname LIKE 'mdt_%';
 staactionname | stasubtype | nspname 
---------------+------------+---------
(0 rows)

